strings/base_meta.h

```diff
@@ -118,17 +118,23 @@ namespace winrt::impl
     };
 
     template <typename T>
+    struct classic_com_guid
```

Comment:
Do we really need to introduce `classic_com_guid` here? And that `static_assert` looks sketchy - like the condition is always going to be false. I think `#error` is preferable in that case.

Reply:
We need to if we want good diagnostics here. A single type alias doesn't leave us much space to use static_assert.

The current strategy is the same one that is being used in impl::producer. We can't use static_assert(false) for the same reason we can't use #error: it would trigger an error even if the user is never using guid_v on a classic COM type.

Reply:
OK, I see what's going on with the `static_assert`, makes sense.

But I still want to see if there's some way to make `classic_com_guid` conditional. Build time with cppwinrt-generated headers is an constant concern, and now guid_v is going to instantiate a second template.

Reply:
Fair enough. I can use `classic_com_guid` only on Clang, and keep the previous simple constexpr variable for other compilers, letting non-MSVC compilers fail on not understanding `__uuidof`.

Reply:
Should be fixed now

Reply:
Thanks!